darwin/dataset/utils.py

```diff
@@ -209,45 +229,48 @@ def split_dataset(
         Keys are the different splits (random, tags, ...) and values are the relative file names
     """
     assert dataset is not None
-    annotation_path = dataset.local_path / "annotations"
+    annotation_path = Path(dataset.local_path / "annotations")
     assert annotation_path.exists()
     annotation_files = list(annotation_path.glob("*.json"))
 
-    # Extract list of classes and create respective files
+    # Prepare the lists folder
     lists_path = dataset.local_path / "lists"
     lists_path.mkdir(parents=True, exist_ok=True)
-    idx_to_classes_polygon = make_class_list(
-        "classes_polygon.txt", annotation_files, lists_path, "polygon", force_resplit, add_background=True
-    )
-    idx_to_classes_tags = make_class_list(
-        "classes_tags.txt", annotation_files, lists_path, "tag", force_resplit
-    )
 
     # Create split id, path and final split paths
-    if val_percentage is None or not 0 < val_percentage < 1.0:
-        raise ValueError(f"The validation percentage is invalid ({val_percentage}). "
-                         f"Must be between 0 and 1.0")
-    if test_percentage is None or not 0 <= test_percentage < 1.0:
-        raise ValueError(f"The test percentage is invalid ({test_percentage}). "
-                         f"Must be between 0 and 1.0")
-    if split_seed is None:
-        raise ValueError(f"The split seed is invalid ({split_seed}).")
-    if not val_percentage + test_percentage < 1.0:
-        raise ValueError(f"The validation ({val_percentage}) + test ({test_percentage}) percentages "
-                         f"must be lower than 1.0 (is {val_percentage + test_percentage})")
-    split_id = f'split_v{int(val_percentage*100)}_t{int(test_percentage*100)}_s{split_seed}'
+    assert val_percentage is not None
+    assert test_percentage is not None
+    assert split_seed is not None
+    assert 0 < val_percentage < 1.0
+    assert 0 <= test_percentage < 1.0
+    assert val_percentage + test_percentage < 1.0
```

Comment:
assert to validate our on invariants, not to validate api consumers input.

Reply:
done